bot/exts/moderation/incidents.py

```diff
@@ -418,10 +536,106 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
+        """
+        Checks if there's any message links in the text content.
+
+        Then passes the the message_link into `make_message_link_embed` to format a
+        embed for it containing information about the link.
+
+        As discord only allows a max of 10 embeds in a single webhook, just send the
+        first 10 embeds and don't care about the rest.
+
+        If no links are found for the message, it logs a trace statement.
+        """
+        message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+
+        if message_links:
+            embeds = []
+            for message_link in message_links[:10]:
+                ctx = await self.bot.get_context(message)
+                embed = await make_message_link_embed(ctx, message_link[0])
+                if embed:
+                    embeds.append(embed)
+
+            return embeds
+
+        log.trace(
+            f"No message links detected on incident message with id {message.id}."
+        )
+
+    async def send_message_link_embeds(
+            self,
+            webhook_embed_list: t.List,
+            message: discord.Message,
+            webhook: discord.Webhook,
+    ) -> t.Optional[int]:
+        """
+        Send Message Link Embeds to #incidents channel.
+
+        Uses the `webhook` passed in as a parameter to send
+        the embeds in the `webhook_embed_list` parameter.
+
+        After sending each webhook it maps the `message.id`
+        to the `webhook_msg_ids` IDs in the async redis-cache.
+        """
+        try:
+            webhook_msg = await webhook.send(
+                embeds=[embed for embed in webhook_embed_list if embed],
+                username=sub_clyde(message.author.name),
+                avatar_url=message.author.avatar_url,
+                wait=True,
+            )
+
+        except discord.DiscordException:
+            log.exception(
+                f"Failed to send message link embed {message.id} to #incidents."
+            )
+
+        else:
+            await self.message_link_embeds_cache.set(message.id, webhook_msg.id)
+            log.trace("Message Link Embed Sent successfully!")
+            return webhook_msg.id
+
+    async def delete_msg_link_embed(self, message_id: int) -> None:
+        """Delete discord message link message found in cache for `message`."""
+        log.trace("Deleting discord links webhook message.")
+        webhook_msg_id = await self.message_link_embeds_cache.get(int(message_id))
+
+        if webhook_msg_id:
+            try:
+                await self.incidents_webhook.delete_message(webhook_msg_id)
+            except discord.errors.NotFound:
+                log.trace(f"Incidents message link embed (`{webhook_msg_id}`) has already been deleted, skipping.")
+
+        await self.message_link_embeds_cache.delete(message_id)
+        log.trace("Successfully deleted discord links webhook message.")
```

Comment:
There's a lot of trace calls here, is it not better if we somehow summarize it in the end of what happened?